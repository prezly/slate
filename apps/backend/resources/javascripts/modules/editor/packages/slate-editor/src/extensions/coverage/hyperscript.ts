/* eslint-disable @typescript-eslint/no-namespace */

import type { CoverageEntry } from '@prezly/sdk';
import { COVERAGE_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-coverage': {
                children?: ReactNode;
                coverage: {
                    id: CoverageEntry['id'];
                };
                uuid: string;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-coverage': { type: COVERAGE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
