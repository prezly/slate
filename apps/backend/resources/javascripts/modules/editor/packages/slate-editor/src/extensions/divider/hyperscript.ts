/* eslint-disable @typescript-eslint/no-namespace */

import { DIVIDER_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-divider': {
                children?: ReactNode;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-divider': { type: DIVIDER_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
