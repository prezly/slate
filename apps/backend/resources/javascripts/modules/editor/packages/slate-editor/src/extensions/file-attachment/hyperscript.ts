/* eslint-disable @typescript-eslint/no-namespace */

import { ATTACHMENT_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { UploadedFile } from '@prezly/uploadcare';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-file-attachment': {
                children?: ReactNode;
                description: string;
                file: UploadedFile;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-file-attachment': { type: ATTACHMENT_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
