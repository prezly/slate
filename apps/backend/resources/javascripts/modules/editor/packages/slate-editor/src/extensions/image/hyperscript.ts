/* eslint-disable @typescript-eslint/no-namespace */

import type { ImageLayout } from '@prezly/slate-types';
import { IMAGE_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { UploadedFile } from '@prezly/uploadcare';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-image': {
                children?: ReactNode;
                file: UploadedFile;
                href: string;
                layout: ImageLayout;
                width: string;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-image': { type: IMAGE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
