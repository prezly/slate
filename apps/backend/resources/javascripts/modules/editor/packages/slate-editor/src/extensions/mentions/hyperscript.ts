/* eslint-disable @typescript-eslint/no-namespace */

import type { VariableNode } from '@prezly/slate-types';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { VARIABLE_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-placeholder-mention': {
                children?: ReactNode;
                key: VariableNode['key'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-placeholder-mention': { type: VARIABLE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
