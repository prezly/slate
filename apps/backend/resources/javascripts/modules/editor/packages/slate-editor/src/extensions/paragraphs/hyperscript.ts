/* eslint-disable @typescript-eslint/no-namespace */

import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
