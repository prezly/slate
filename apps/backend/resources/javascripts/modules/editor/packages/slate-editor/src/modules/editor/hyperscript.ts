/* eslint-disable @typescript-eslint/no-namespace */

import {
    BULLETED_LIST_NODE_TYPE,
    DIVIDER_NODE_TYPE,
    LIST_ITEM_NODE_TYPE,
    LIST_ITEM_TEXT_NODE_TYPE,
    NUMBERED_LIST_NODE_TYPE,
    PARAGRAPH_NODE_TYPE,
    QUOTE_NODE_TYPE,
} from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-divider': {
                children?: ReactNode;
            };
            'h-li': {
                children?: ReactNode;
            };
            'h-li-text': {
                children?: ReactNode;
            };
            'h-ol': {
                children?: ReactNode;
            };
            'h-p': {
                children?: ReactNode;
            };
            'h-ul': {
                children?: ReactNode;
            };
            'h:blockquote': {
                children?: ReactNode;
            };
            'h:text': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-p': { type: PARAGRAPH_NODE_TYPE },
        'h-divider': { type: DIVIDER_NODE_TYPE },
        'h-ol': { type: NUMBERED_LIST_NODE_TYPE },
        'h-ul': { type: BULLETED_LIST_NODE_TYPE },
        'h-li': { type: LIST_ITEM_NODE_TYPE },
        'h-li-text': { type: LIST_ITEM_TEXT_NODE_TYPE },
        'h:blockquote': { type: QUOTE_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
        'h:text': createText,
    },
});
