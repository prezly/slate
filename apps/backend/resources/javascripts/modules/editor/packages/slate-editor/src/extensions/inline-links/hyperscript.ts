/* eslint-disable @typescript-eslint/no-namespace,@typescript-eslint/no-unused-vars */
import { LINK_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { createEditor as createSlateEditor } from 'slate';
import { createHyperscript, createEditor as createEditorFactory } from 'slate-hyperscript';

import { createEditor } from '#modules/editor';

import { InlineLinksExtension } from './InlineLinksExtension';

// This allows tests to include Slate Nodes written in JSX without TypeScript complaining.
declare global {
    namespace JSX {
        interface IntrinsicElements {
            [elemName: string]: any; // eslint-disable-line
        }
    }
}

const extensions = [InlineLinksExtension()];

export const hyperscript = createHyperscript({
    elements: {
        link: { type: LINK_NODE_TYPE },
        paragraph: { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        editor: createEditorFactory(() => createEditor(createSlateEditor(), () => extensions)),
    },
});
