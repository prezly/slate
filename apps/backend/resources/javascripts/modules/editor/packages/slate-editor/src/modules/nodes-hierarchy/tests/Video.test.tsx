/** @jsx hyperscript */

import { Editor } from 'slate';

import { hyperscript } from '../../../hyperscript';

describe('nodes-hierarchy / Video', () => {
    it('should be kept after normalization', function () {
        const editor = (
            <editor>
                <h:video
                    url="https://www.youtube.com/watch?v=FYH9n37B7Yw&t=1s"
                    uuid="863fd1d6-fcac-46f7-b5c2-66c459b3f3de"
                    layout="expanded"
                    oembed={{
                        author_url: 'https://www.youtube.com/channel/UCjBp_7RuDBUYbd1LegWEJ8g',
                        cache_age: 86400,
                        description:
                            'Your ultimate Horizon adventure awaits! Explore the vibrant and ever-evolving open world landscapes of Mexico with limitless, fun driving action in hundreds of the worldâ€™s greatest cars. \n\nPlay Forza Horizon 5 day one with Xbox Game Pass. \nhttps://www.xbox.com/games/forza-horizon-5\n\nAvailable on Xbox Series X|S, Xbox One, Windows 10 PC, and Steam.â€‹\n\nSubscribe to Xbox ðŸŽ® https://xbx.lv/2EEjmaR\n\nFollow Xbox on other channels:\nFacebook: https://www.facebook.com/Xbox\nTwitter: https://www.twitter.com/Xbox\nInstagram: https://www.instagram.com/Xbox',
                        html: '<div><div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;"><iframe src="//cdn.iframe.ly/api/iframe?url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DFYH9n37B7Yw%26t%3D1s&key=8fe6cdec03482ac31f27a6ae8ea2fb3f" style="top: 0; left: 0; width: 100%; height: 100%; position: absolute; border: 0;" allowfullscreen scrolling="no" allow="accelerometer *; clipboard-write *; encrypted-media *; gyroscope *; picture-in-picture *;"></iframe></div></div>',
                        provider_name: 'YouTube',
                        screenshot_url:
                            'https://avatars-cdn.prezly.com/embed/aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g.dj1GWUg5bjM3QjdZdyZ0PTFz/a29634b1924c9a0db6fce8aead3010a3578190e06ceb2837d61f33cbd2194d40?v=1583930760',
                        thumbnail_height: 720,
                        thumbnail_url: 'https://i.ytimg.com/vi/FYH9n37B7Yw/maxresdefault.jpg',
                        thumbnail_width: 1280,
                        title: 'Forza Horizon 5 Official Announce Trailer',
                        type: 'video',
                        url: 'https://www.youtube.com/watch?v=FYH9n37B7Yw',
                        version: '1.0',
                    }}
                />
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:video
                    url="https://www.youtube.com/watch?v=FYH9n37B7Yw&t=1s"
                    uuid="863fd1d6-fcac-46f7-b5c2-66c459b3f3de"
                    layout="expanded"
                    oembed={{
                        author_url: 'https://www.youtube.com/channel/UCjBp_7RuDBUYbd1LegWEJ8g',
                        cache_age: 86400,
                        description:
                            'Your ultimate Horizon adventure awaits! Explore the vibrant and ever-evolving open world landscapes of Mexico with limitless, fun driving action in hundreds of the worldâ€™s greatest cars. \n\nPlay Forza Horizon 5 day one with Xbox Game Pass. \nhttps://www.xbox.com/games/forza-horizon-5\n\nAvailable on Xbox Series X|S, Xbox One, Windows 10 PC, and Steam.â€‹\n\nSubscribe to Xbox ðŸŽ® https://xbx.lv/2EEjmaR\n\nFollow Xbox on other channels:\nFacebook: https://www.facebook.com/Xbox\nTwitter: https://www.twitter.com/Xbox\nInstagram: https://www.instagram.com/Xbox',
                        html: '<div><div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;"><iframe src="//cdn.iframe.ly/api/iframe?url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DFYH9n37B7Yw%26t%3D1s&key=8fe6cdec03482ac31f27a6ae8ea2fb3f" style="top: 0; left: 0; width: 100%; height: 100%; position: absolute; border: 0;" allowfullscreen scrolling="no" allow="accelerometer *; clipboard-write *; encrypted-media *; gyroscope *; picture-in-picture *;"></iframe></div></div>',
                        provider_name: 'YouTube',
                        screenshot_url:
                            'https://avatars-cdn.prezly.com/embed/aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g.dj1GWUg5bjM3QjdZdyZ0PTFz/a29634b1924c9a0db6fce8aead3010a3578190e06ceb2837d61f33cbd2194d40?v=1583930760',
                        thumbnail_height: 720,
                        thumbnail_url: 'https://i.ytimg.com/vi/FYH9n37B7Yw/maxresdefault.jpg',
                        thumbnail_width: 1280,
                        title: 'Forza Horizon 5 Official Announce Trailer',
                        type: 'video',
                        url: 'https://www.youtube.com/watch?v=FYH9n37B7Yw',
                        version: '1.0',
                    }}
                />
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toMatchSnapshot();
        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });
});
