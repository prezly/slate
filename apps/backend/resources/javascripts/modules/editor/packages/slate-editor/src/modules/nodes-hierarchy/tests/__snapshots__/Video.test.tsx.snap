// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodes-hierarchy / Video should be kept after normalization 1`] = `
[
  {
    "children": [
      {
        "text": "",
      },
    ],
    "layout": "expanded",
    "oembed": {
      "author_url": "https://www.youtube.com/channel/UCjBp_7RuDBUYbd1LegWEJ8g",
      "cache_age": 86400,
      "description": "Your ultimate Horizon adventure awaits! Explore the vibrant and ever-evolving open world landscapes of Mexico with limitless, fun driving action in hundreds of the worldâ€™s greatest cars. 

Play Forza Horizon 5 day one with Xbox Game Pass. 
https://www.xbox.com/games/forza-horizon-5

Available on Xbox Series X|S, Xbox One, Windows 10 PC, and Steam.â€‹

Subscribe to Xbox ðŸŽ® https://xbx.lv/2EEjmaR

Follow Xbox on other channels:
Facebook: https://www.facebook.com/Xbox
Twitter: https://www.twitter.com/Xbox
Instagram: https://www.instagram.com/Xbox",
      "html": "<div><div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;"><iframe src="//cdn.iframe.ly/api/iframe?url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DFYH9n37B7Yw%26t%3D1s&key=8fe6cdec03482ac31f27a6ae8ea2fb3f" style="top: 0; left: 0; width: 100%; height: 100%; position: absolute; border: 0;" allowfullscreen scrolling="no" allow="accelerometer *; clipboard-write *; encrypted-media *; gyroscope *; picture-in-picture *;"></iframe></div></div>",
      "provider_name": "YouTube",
      "screenshot_url": "https://avatars-cdn.prezly.com/embed/aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g.dj1GWUg5bjM3QjdZdyZ0PTFz/a29634b1924c9a0db6fce8aead3010a3578190e06ceb2837d61f33cbd2194d40?v=1583930760",
      "thumbnail_height": 720,
      "thumbnail_url": "https://i.ytimg.com/vi/FYH9n37B7Yw/maxresdefault.jpg",
      "thumbnail_width": 1280,
      "title": "Forza Horizon 5 Official Announce Trailer",
      "type": "video",
      "url": "https://www.youtube.com/watch?v=FYH9n37B7Yw",
      "version": "1.0",
    },
    "type": "video",
    "url": "https://www.youtube.com/watch?v=FYH9n37B7Yw&t=1s",
    "uuid": "863fd1d6-fcac-46f7-b5c2-66c459b3f3de",
  },
]
`;
